src/hooks/useProfileUpdate.ts

```diff
@@ -0,0 +1,47 @@
+import { useState, useRef, ChangeEvent } from "react";
+import { updateProfile } from "firebase/auth";
+import { doc, updateDoc } from "firebase/firestore";
+import { db } from "../store/auth/useAuthStore";
+import { User } from "firebase/auth";
+
+export const useUpdateProfile = (user: User | null) => {
+  const [loading, setLoading] = useState(false);
+  const [successMessage, setSuccessMessage] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+  const [photoURL, setPhotoURL] = useState(user?.photoURL || "");
+
+  const handleUpdateProfile = async (phoneNumber: string) => {
+    setLoading(true);
+    setErrorMessage("");
+    setSuccessMessage("");
+
+    try {
+      if (user) {
+        // Оновлюємо фото профілю
+        await updateProfile(user, { photoURL });
```

Comment:
Update profile logic should be in authStore, where you handle user data, components/hooks should not use firebase at all, only store has exes to firestore/firebase